src/yoroi-wallets/cardano/nfts.ts

```diff
@@ -2,17 +2,23 @@ import {getAssetFingerprint} from '../../legacy/format'
 import {AssetMetadata, NftMetadata, YoroiNft} from '../types'
 import {asciiToHex} from './api/utils'
 
-export const convertNft = (assetMetadata: AssetMetadata, storageUrl: string): YoroiNft => {
+export const convertNfts = (assetMetadata: AssetMetadata, storageUrl: string): YoroiNft[] => {
   const policyId = Object.keys(assetMetadata)[0]
-  const assetNameKey = Object.keys(assetMetadata[policyId])[0]
-  const metadata: NftMetadata = assetMetadata[policyId][assetNameKey]
-  const assetNameHex = asciiToHex(assetNameKey)
+  const allMetadataKeys = Object.keys(assetMetadata[policyId])
+  return allMetadataKeys.map((name) => {
```

Comment:
I think the endpoint is returning the entire minting metadata from the last mint transaction, it can be narrowed down to only the subject in the query, anyways, new API design this is the first endpoint to tackle.